### DISCUSSION TITLE
Build Failed with libyaml & psych on Almalinux 8.6

### DISCUSSION THREAD

**Discussion by daBee** [rbenv/ruby-build#2192] [Build failures] [answered]
I am using `ruby-build 20220630` and attempting version `3.2.2` on `Almalinux 8.6`.  It seems `psych` is a bit out of date.

```
[Tue May 02 19:34:33 user3@server_f ~] yum search psych
=========================== Name Matched: psych ==================================
rubygem-psych.x86_64 : A libyaml wrapper for Ruby
rubygem-psych.i686 : A libyaml wrapper for Ruby
``` 

```
/tmp/ruby-build.20230502193059.45504.2taALp/ruby-3.2.2/lib/yaml.rb:3: warning: It seems your ruby installation is missing psych (for YAML output).
To eliminate this warning, please install libyaml and reinstall your ruby.
uh-oh! RDoc had a problem:
cannot load such file -- psych

run with --debug for full backtrace
make: *** [uncommon.mk:598: rdoc] Error 1

BUILD FAILED (AlmaLinux 8.7 using ruby-build 20230428)

Inspect or clean up the working tree at /tmp/ruby-build.20230502193059.45504.2taALp
Results logged to /tmp/ruby-build.20230502193059.45504.log

Last 10 log lines:
*** Fix the problems, then remove these directories and try again if you want.
make[1]: Leaving directory '/tmp/ruby-build.20230502193059.45504.2taALp/ruby-3.2.2'
Generating RDoc documentation
/tmp/ruby-build.20230502193059.45504.2taALp/ruby-3.2.2/lib/yaml.rb:3: warning: It seems your ruby installation is missing psych (for YAML output).
To eliminate this warning, please install libyaml and reinstall your ruby.
uh-oh! RDoc had a problem:
cannot load such file -- psych
```

Is there a way around this?  Any insight appreciated.

**Comment by mislav** [comment]
Hi, per our [suggested build environment](https://github.com/rbenv/ruby-build/wiki#suggested-build-environment), you should install `libyaml-devel` with your package manager, not "psych".

**Comment by daBee** [comment]
Added both.  Buuuuut, still issues.  I'm actually migrating from `8.7` (the actual version) to 9.x.

**Reply by mislav** [reply]
If you observe the `Results logged to` line, you will see a file where the full log is. You can inspect that, or share here as a Gist. The log may contain more information.

Additionally, if you cd into the directory printed in the `Inspect or clean up the working tree at` line, you can find `mkmf.log` files under various `ext` subdirectories. If a Ruby extension failed to compile, those log files may contain additional information. If you find something, you are welcome to share. Thanks!

**Comment by daBee** [comment]
Ya it's reporting the same issue about `psych`.  I'm upgrading to `9.2` but I thought I'd give it a try:

```
Generating RDoc documentation
/tmp/ruby-build.20230511134331.108518.FhYCYp/ruby-3.2.2/lib/yaml.rb:3: warning: It seems your ruby installation is missing psych (for YAML output).
To eliminate this warning, please install libyaml and reinstall your ruby.
uh-oh! RDoc had a problem:
cannot load such file -- psych

run with --debug for full backtrace
make: *** [uncommon.mk:598: rdoc] Error 1
```

**Comment by dxg** [comment, marked as answer]
For anyone struggling to install `libyaml-devel` in AlmaLinux 8 or 9, `libyaml-devel` is part of their [CodeReady Builder/PowerTools](https://wiki.almalinux.org/repos/AlmaLinux.html) repo which has to be enabled:

AlmaLinux 8:
```sh
dnf config-manager --set-enabled powertools
```
AlmaLinux 9:
```sh
dnf config-manager --set-enabled crb
```
Then:
```sh
dnf install libyaml-devel
```
after which Ruby 3.2 compiles successfully.
